node-native/src/lib.rs

```diff
@@ -0,0 +1,413 @@
+
+    #[cfg(target_os = "android")]
+    {
+        std::env::var("LUMINA_DATA_DIR")
+            .map(PathBuf::from)
+            .map_err(|e| LuminaError::StorageError {
+                msg: format!("Could not get LUMINA_DATA_DIR: {}", e),
+            })
+    }
+
+    #[cfg(not(any(target_os = "ios", target_os = "android")))]
+    {
+        Err(LuminaError::StorageError {
+            msg: "Unsupported platform".to_string(),
+        })
+    }
+}
+
+/// Represents all possible errors that can occur in the LuminaNode.
+#[derive(Error, Debug, uniffi::Error)]
+pub enum LuminaError {
+    /// Error returned when trying to perform operations on a node that isn't running
+    #[error("Node is not running")]
+    NodeNotRunning,
+
+    /// Error returned when network operations fail
+    #[error("Network error: {msg}")]
+    NetworkError {
+        /// Description of the network error
+        msg: String,
+    },
+
+    /// Error returned when storage operations fail
+    #[error("Storage error: {msg}")]
+    StorageError {
+        /// Description of the storage error
+        msg: String,
+    },
+
+    /// Error returned when trying to start a node that's already running
+    #[error("Node is already running")]
+    AlreadyRunning,
+
+    /// Error returned when a mutex lock operation fails
+    #[error("Lock error")]
+    LockError,
+
+    /// Error returned when a hash string is invalid or malformed
+    #[error("Invalid hash format: {msg}")]
+    InvalidHash {
+        /// Description of why the hash is invalid
+        msg: String,
+    },
+
+    /// Error returned when a header is invalid or malformed
+    #[error("Invalid header format: {msg}")]
+    InvalidHeader {
+        /// Description of why the header is invalid
+        msg: String,
+    },
+
+    /// Error returned when storage initialization fails
+    #[error("Storage initialization failed: {msg}")]
+    StorageInit {
+        /// Description of why storage initialization failed
+        msg: String,
+    },
+}
+
+impl From<NodeError> for LuminaError {
+    fn from(error: NodeError) -> Self {
+        LuminaError::NetworkError {
+            msg: error.to_string(),
+        }
+    }
+}
+
+/// The main Lumina node that manages the connection to the Celestia network.
+#[derive(Object)]
+pub struct LuminaNode {
+    node: Arc<Mutex<Option<Node<RedbBlockstore, RedbStore>>>>,
+    network: Network,
+    events_subscriber: Arc<Mutex<Option<EventSubscriber>>>,
+}
+
+#[uniffi::export(async_runtime = "tokio")]
+impl LuminaNode {
+    /// Sets a new connection to the Lumina node for the specified network.
+    #[uniffi::constructor]
+    pub fn new(network: Network) -> Result<Arc<Self>> {
+        Ok(Arc::new(Self {
+            node: Arc::new(Mutex::new(None)),
+            network,
+            events_subscriber: Arc::new(Mutex::new(None)),
+        }))
+    }
+
+    /// Starts the Lumina node. Returns true if successfully started.
+    pub async fn start(&self) -> Result<bool> {
```

Comment:
could we accept here some config struct that will allow for more control / tuning node?

Reply:
sure, like the wasm `NodeStartConfig`? Or what configurations do you think of?